### PR TITLE
ddl/ingest: fix local flush logic for multi-schema change

### LINKED ISSUE #50990: the speed of fast-reorg add multiple indexes with multi-schema change is slower than fast-reorg=0
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

1. enable fast-reorg
2. disable tidb_enable_dist_task 
3. alter table items add index `index1` ((cast(`col1` as char(253) array))), add index `index2` ((cast(`col2` as char(32) array)));

### 2. What did you expect to see? (Required)
add index successfully and speed is faster than fast-reorg=0
### 3. What did you see instead (Required)
the speed is slower than fast-reorg=0
### 4. What is your TiDB version? (Required)
7.6.0

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close #50990

Problem Summary:

For multi-schema change like `add index i1, add index i2`, only one index is imported when `ddl_disk_quota` is reached:

```
/tmp/tidb/tmp_ddl-4000/108$ du -hs ./*
16K     ./35a62bba-3aab-5322-b59b-c5bc8baf9c9d
4.0K    ./35a62bba-3aab-5322-b59b-c5bc8baf9c9d.sst
16K     ./8d2242cb-c9a7-5915-a93a-45927b24759d
4.0G    ./8d2242cb-c9a7-5915-a93a-45927b24759d.sst

/tmp/tidb/tmp_ddl-4000/108$ du -hs ./*
16K     ./35a62bba-3aab-5322-b59b-c5bc8baf9c9d
4.0K    ./35a62bba-3aab-5322-b59b-c5bc8baf9c9d.sst
16K     ./8d2242cb-c9a7-5915-a93a-45927b24759d
16G     ./8d2242cb-c9a7-5915-a93a-45927b24759d.sst

/tmp/tidb/tmp_ddl-4000/108$ du -hs ./*
16K     ./35a62bba-3aab-5322-b59b-c5bc8baf9c9d
4.0K    ./35a62bba-3aab-5322-b59b-c5bc8baf9c9d.sst
16K     ./8d2242cb-c9a7-5915-a93a-45927b24759d
18G     ./8d2242cb-c9a7-5915-a93a-45927b24759d.sst
```

As a result, almost every region batch triggers import, making it very slow.

### What changed and how does it work?

Import all indexes if `ddl_disk_quota` is reached.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/reorg.go [hunk 1]
```diff
@@ -291,7 +291,11 @@ func overwriteReorgInfoFromGlobalCheckpoint(w *worker, sess *sess.Session, job *
 	if ok {
 		// We create the checkpoint manager here because we need to wait for the reorg meta to be initialized.
 		if bc.GetCheckpointManager() == nil {
-			mgr, err := ingest.NewCheckpointManager(w.ctx, bc, w.sessPool, job.ID, reorgInfo.currElement.ID)
+			elemIDs := make([]int64, 0, len(reorgInfo.elements))
+			for _, elem := range reorgInfo.elements {
+				elemIDs = append(elemIDs, elem.ID)
+			}
```
Comment (ywqzzy) [inline pkg/ddl/reorg.go:297]: Also found the code in executeDistTask, may be wrap a function

File: pkg/ddl/ingest/checkpoint.go [hunk 2]
```diff
@@ -348,7 +362,7 @@ func (s *CheckpointManager) updateCheckpoint() error {
 	defer s.sessPool.Put(sessCtx)
 	ddlSess := sess.NewSession(sessCtx)
 	err = ddlSess.RunInTxn(func(se *sess.Session) error {
-		template := "update mysql.tidb_ddl_reorg set reorg_meta = %s where job_id = %d and ele_id = %d and ele_type = %s;"
+		template := "update mysql.tidb_ddl_reorg set reorg_meta = %s where job_id = %d and ele_type = %s;"
```
Comment (ywqzzy) [inline pkg/ddl/ingest/checkpoint.go:365]: Then for all different indexes, the reorg_meta are the same?
Is it expected?

Comment (tangenta) [reply pkg/ddl/ingest/checkpoint.go:365]: Yes. The checkpoint only records table row handles.

File: pkg/ddl/ingest/checkpoint.go [hunk 3]
```diff
@@ -194,6 +194,20 @@ func (s *CheckpointManager) UpdateCurrent(taskID int, added int) error {
 	return nil
 }
 
+func (s *CheckpointManager) tryFlushAllIndexes(mode FlushMode) (flushed, imported bool, err error) {
+	allFlushed := true
+	allImported := true
+	for _, idxID := range s.indexIDs {
+		flushed, imported, err := s.flushCtrl.Flush(idxID, mode)
+		if err != nil {
+			return flushed, false, err
```
Comment (wjhuang2016) [inline pkg/ddl/ingest/checkpoint.go:203]: allFlushed = allFlushed && flushed
return allFlushed, false, err ?

**Review by wjhuang2016 — APPROVED** [review]

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Pull request merged**